### ISSUE TITLE
[1.3.13] Add a validate command and/or dry run feature

### ISSUE THREAD

**Issue by schickling** [graphcool/prisma#659]
<a href="https://github.com/kbrandwijk"><img src="https://avatars1.githubusercontent.com/u/852069?v=4" align="left" width="96" height="96" hspace="10"></img></a> **Issue by [kbrandwijk](https://github.com/kbrandwijk)**
_Tuesday Sep 12, 2017 at 14:29 GMT_
_Originally opened as https://github.com/graphcool/cli/issues/235_

----

I don't know if it's mentioned already somewhere, or if it's similar to doing a 'dry run', but it would be nice to be able to run a validate command that would check:
- Project file syntax
- Permission and function types found in schema
- Graphql queries/PQ/SE SDL valid
- ...?

**Comment by schickling** [comment]
<a href="https://github.com/Merott"><img src="https://avatars3.githubusercontent.com/u/1757708?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [Merott](https://github.com/Merott)**
_Monday Sep 25, 2017 at 10:55 GMT_

----

I was going to open a feature request for a dry-run option, and I came across this.

My use case is that I want to be able to dry-run the `deploy` command as part of my CI build process, in order to ensure that the deployment would succeed once a feature branch is merged into master.

At the moment `graphcool diff` is the closest thing we have to this, as it achieves validation, but it doesn't verify that the deployment would succeed. For example, if it's a destructive deployment because of a removed field or type, `deploy` would fail unless `--force` is used, but not `diff`.

I would propose a `--dry-run` option for `graphcool deploy`. Should I open a separate issue for it?

**Comment by schickling** [comment]
<a href="https://github.com/kbrandwijk"><img src="https://avatars1.githubusercontent.com/u/852069?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [kbrandwijk](https://github.com/kbrandwijk)**
_Monday Sep 25, 2017 at 11:27 GMT_

----

@Merott Thanks for the valuable addition. Especially the destructive deployment part.

**Comment by schickling** [comment]
<a href="https://github.com/marktani"><img src="https://avatars1.githubusercontent.com/u/1780597?v=4" align="left" width="48" height="48" hspace="10"></img></a> **Comment by [marktani](https://github.com/marktani)**
_Monday Sep 25, 2017 at 11:27 GMT_

----

I think this issue is great to discuss a dry run feature 🙂 

> For example, if it's a destructive deployment because of a removed field or type, deploy would fail unless --force is used, but not diff.

I thought `diff` is already covering this, but that's a great point 👌

**Comment by marktani** [comment]
This issue has been moved to [graphcool/graphcool-framework](https://github.com/graphcool/graphcool-framework/).
